elk/evaluation/evaluate.py

```diff
@@ -31,7 +32,7 @@ def execute(self, highlight_color: Color = "cyan"):
     @torch.inference_mode()
     def apply_to_layer(
         self, layer: int, devices: list[str], world_size: int
-    ) -> dict[str, pd.DataFrame]:
+    ) -> tuple[dict[str, pd.DataFrame], list[dict]]:
```

Comment:
Would suggest to add type aliases, list of dict is not very nice to read

`LayerOutput = list[dict]`

Reply:
And make a new dataclass `LayerResult` that encapsulates the whole output instead of simply returning a tuple